src/ImpurityDensities/BouleImpurityDensities/SplineBouleImpurityDensity.jl

```diff
@@ -0,0 +1,36 @@
+"""
+    struct SplineBouleImpurityDensity{N, T <: SSDFloat} <: AbstractImpurityDensity{T}
+
+Impurity density model which assumes an arbitrary impurity density in z (boule coordinates) defined by cubic spline interpolation between given points.
+
+## Fields
+* `z::SVector{N, T}`: z coordinates (in boule coordinates) where the impurity density is defined.
+* `ρ::SVector{N, T}`: impurity density values at the given z coordinates.
+* `spline::Interpolations.Extrapolation{T}`: cubic spline interpolation of the impurity density values.
+* `det_z0::T`: z coordinate of the detector origin in boule coordinates. The z-direction of the detector is opposite to the z-direction of the boule coordinates.
+"""
+struct SplineBouleImpurityDensity{N, T <: SSDFloat} <: AbstractImpurityDensity{T}
+    z::SVector{N, T}
+    ρ::SVector{N, T}
+    spline::Interpolations.Extrapolation{T}
+    det_z0::T
+end
+
+function SplineBouleImpurityDensity{T}(z::Vector{<:Number}, ρ::Vector{<:Number}, det_z0::Number) where {T}
```

Comment:
```suggestion
function SplineBouleImpurityDensity{T}(z::AbstractVector{<:Number}, ρ::AbstractVector{<:Number}, det_z0::Number) where {T}
```
(and maybe `RealQuantity` instead of `Number` (?))